test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using NuGet.VisualStudio.Internal.Contracts;
+using Test.Utility;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    [Collection(MockedVS.Collection)]
+    public class NuGetPackageSearchServiceTests : MockedVSCollectionTests
+    {
+        private readonly SourceRepository _sourceRepository;
+        private readonly IEnumerable<IPackageReferenceContextInfo> _installedPackages;
+        private readonly IReadOnlyCollection<IProjectContextInfo> _projects;
+
+        public NuGetPackageSearchServiceTests(GlobalServiceProvider globalServiceProvider)
+            : base(globalServiceProvider)
+        {
+            _installedPackages = new List<IPackageReferenceContextInfo>();
+            _projects = new List<IProjectContextInfo>
+            {
+                new ProjectContextInfo(
+                    Guid.NewGuid().ToString(),
+                    ProjectModel.ProjectStyle.PackageReference,
+                    NuGetProjectKind.PackageReference)
+            };
+            var testFeedUrl = "https://testsource.com/v3/index.json";
```

Comment:
better to use a domain name ending in `.test`, to prevent accidentally making requests to hosts in case someone registers that domain.

Reply:
will fix with https://github.com/NuGet/Home/issues/10151